### DISCUSSION TITLE
Plotting stability on MacOS

### DISCUSSION THREAD

**Discussion by ICYou** [Chia-Network/chia-blockchain#1489] [Q&A]
I'm currently experimenting with plotting. I have done that on multiple mac devices, using external temporary storage, but with all of them, at some point I have seen the system shut down due to "a system error". Also I have seen the "Chia Helper (Renderer)" process taking up to 5GB of RAM, causing the system to swap. 

**What I'm wondering is, what should I take into account to ensure that the plotter runs smoothly?**

I have some ideas that perhaps could be a cause for instability, or could help with stability:
- Chia taking up too many resources/error on a bottleneck.
- Plotting using temp storage on external drive is less stable.
- Plotting using GUI is less stable than through command line.
- Plotting with a full mac os system is less stable than a barebones linux system would be.

Perhaps anyone has thoughts on this or on my plotting configurations?

Mac mini 2018, i7 32gb ram, 1TB SSD
- Temp storage on: Samsung x5 thunderbolt3 1TB ssd
- Two plots in parallel (6GB RAM, 8 threads, no bitfield)
- Saving to internal SSD

MB Pro 2018, i7 16gb ram, 256GB SSD
- Temp storage on: Samsung x5 thunderbolt3 1TB ssd
- Two plots in parallel (5GB RAM, 4 threads, no bitfield)
- Saving to external HDD

MB Pro 2013, i7 8gb ram, 256GB SSD
- Temp storage on: HDD over usb 3.1
- Single plot (3500GB RAM, 4 threads, with bitfield)
- Saving to network storage

**Comment by tangdasan** [comment]
How fast are you running？

**Comment by cccat6** [comment]
My guess
1. Is the power supply for the external drive enough? Laptop usually does not have satisfactory output power via USB.
2. MacOS love to swap. You know, even though there is much free memory left in the system. MacOS will swap anyway.

**Comment by ICYou** [comment]
I have completely moved to plotting on the mac mini with a clean install of macOS.
Since then I haven't had problems with plotting. The machine has been steadily outputting plots since.
I am not sure what has been the cause with the other machines.
